balancer/balancer.go

```diff
@@ -47,6 +47,14 @@ func Register(b Builder) {
 	m[strings.ToLower(b.Name())] = b
 }
 
+// UnregisterForTesting deletes the balancer with the given name from the
+// balancer map.
+//
+// This function is not thread-safe. And should only be used for testing.
+func UnregisterForTesting(name string) {
```

Comment:
Can we move this into `internal`, please?  (I.e. store the map itself there.)  I want to avoid cluttering our external API with implementation details or testing functions at all costs.

Reply:
Added an internal function pointer.